src/upload/form/modules_worker.rs

```diff
@@ -53,6 +55,8 @@ fn add_metadata(mut form: Form, assets: &ModulesAssets) -> Result<Form> {
         bindings: assets.bindings(),
         migrations: assets.migration.clone(),
         usage_model: assets.usage_model,
+        compatibility_date: assets.compatibility_date.clone(),
+        compatibility_flags: assets.compatibility_flags.clone(),
```

Comment:
Unrelated to your change, but it bugs me that there are so many spurious clones here, `Metadata` could hold references instead of owned values. `build_form` should take an owned value too since it's always used with `&owned_assets` and clones the fields.